ServerManager/api/editGameSession.php

```diff
@@ -34,8 +36,15 @@
     $api->setPayLoad([$action => $gamesession->$action()]);
 }
 
-// ready to do final actual update
-$gamesession->edit();
+// alternative to recreate function in GameSession class
+if ($action == 'recreate') {
+    $gamesession->setToLoading();
+    $gamesession->edit();
+    SymfonyToLegacyHelper::getInstance()->getMessageBus()->dispatch(new GameListCreationMessage($gamesession->id));
+} else {
```

Comment:
I wonder if this could be replace 
ServerManager/classes/GameSession.php 's
recreate method ?